src/terraingenerator/main.cpp

```diff
@@ -360,110 +498,30 @@ int main(int argc, char *argv[])
-                }
-            }
-        }
-    }
-
-
-
-    terrainList.clear();
-    terrainList.append(terrains["PlowedSoil"]);
-    terrainList.append(terrains["Grass"]);
-    foreach (Terrain *topLeft, terrainList) {
-        foreach (Terrain *topRight, terrainList) {
-            foreach (Terrain *bottomLeft, terrainList) {
-                foreach (Terrain *bottomRight, terrainList) {
-                    process.append(TileTerrainNames(topLeft->name(),
-                                                    topRight->name(),
-                                                    bottomLeft->name(),
-                                                    bottomRight->name()));
-                }
-            }
-        }
-    }
-
-    terrainList.clear();
-    terrainList.append(terrains["Dirt"]);
-    terrainList.append(terrains["Hole"]);
-    terrainList.append(terrains["Grass"]);
-    foreach (Terrain *topLeft, terrainList) {
-        foreach (Terrain *topRight, terrainList) {
-            foreach (Terrain *bottomLeft, terrainList) {
-                foreach (Terrain *bottomRight, terrainList) {
-                    process.append(TileTerrainNames(topLeft->name(),
-                                                    topRight->name(),
-                                                    bottomLeft->name(),
-                                                    bottomRight->name()));
-                }
-            }
-        }
-    }
-
-    terrainList.clear();
-    terrainList.append(terrains["Sand"]);
-    terrainList.append(terrains["Water"]);
-    foreach (Terrain *topLeft, terrainList) {
-        foreach (Terrain *topRight, terrainList) {
-            foreach (Terrain *bottomLeft, terrainList) {
-                foreach (Terrain *bottomRight, terrainList) {
-                    process.append(TileTerrainNames(topLeft->name(),
-                                                    topRight->name(),
-                                                    bottomLeft->name(),
-                                                    bottomRight->name()));
+        // Construct a vector with all terrain combinations to process
+        foreach (Terrain *topLeft, terrainList) {
+            foreach (Terrain *topRight, terrainList) {
+                foreach (Terrain *bottomLeft, terrainList) {
+                    foreach (Terrain *bottomRight, terrainList) {
+                        //qDebug()<<topLeft->name()<<"\t"<<topRight->name()<<"\t"<<bottomLeft->name()<<"\t"<<bottomRight->name();
+                        process.append(TileTerrainNames(topLeft->name(),
+                                                        topRight->name(),
+                                                        bottomLeft->name(),
+                                                        bottomRight->name()));
+                    }
                 }
             }
         }
+        terrainList.clear();
```

Comment:
Instead of reusing and clearing the same `terrainList` every time, please just make the variable more local.